File: market/src/main/java/com/wanted/market/member/controller/MemberController.java
package com.wanted.market.member.controller;

import com.wanted.market.member.dto.MemberRequestDto;
import com.wanted.market.member.dto.MemberDetailResponseDto;
import com.wanted.market.member.service.MemberService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/member")
public class MemberController {

    private final MemberService memberService;

    public MemberController(MemberService memberService) {
        this.memberService = memberService;
    }

    @PostMapping("/join")
    public ResponseEntity<MemberDetailResponseDto> join(@RequestBody MemberRequestDto memberRequestDto){
        MemberDetailResponseDto newMember = memberService.join(memberRequestDto);
        return ResponseEntity.ok(newMember);
    }
}

File: api/src/main/kotlin/wanted/market/api/domain/member/controller/MemberController.kt
package wanted.market.api.domain.member.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import wanted.market.api.domain.member.service.MemberService
import wanted.market.api.domain.member.dto.RegisterMemberRequest
import wanted.market.api.domain.member.dto.LoginMemberRequest

@RestController
@RequestMapping("/auth")
class MemberController(
    @Autowired private val memberService: MemberService
) {
    @PostMapping("/register")
    fun registerMember(
        @RequestBody registerMemberRequest: RegisterMemberRequest
    ): ResponseEntity<Any> {
        memberService.registerMember(registerMemberRequest)
        return ResponseEntity.ok().build()
    }

    @PostMapping("/login")
    fun loginMember(
      @RequestBody loginMemberRequest: LoginMemberRequest
    ): ResponseEntity<Any> {
        memberService.loginMember(loginMemberRequest)
        return ResponseEntity.ok().build()
    }
}
